---
title: C6318
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6318
helpviewer_keywords:
- C6318
ms.assetid: 3284a83e-bb8e-461c-adcc-cfc66ceea05e
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 1feeb41725f02985aa35f94815578c5c216ba90f
ms.sourcegitcommit: 535ef05b1e553f0fc66082cd2e0998817eb2a56a
ms.translationtype: MT
ms.contentlocale: pl-PL
ms.lasthandoff: 10/07/2019
ms.locfileid: "72013004"
---
# <a name="c6318"></a>C6318
Ostrzeżenie C6318: Źle zdefiniowana __try/@no__t -0 _Except: Użycie stałej EXCEPTION_CONTINUE_SEARCH lub innej stałej, która daje w wyniku wartość zero w wyrażeniu filtru wyjątków. Kod w bloku obsługi wyjątków nie jest wykonywany

 To ostrzeżenie wskazuje, że jeśli wystąpi wyjątek w chronionym bloku tego programu obsługi wyjątków strukturalnych, wyjątek nie zostanie obsłużony, ponieważ stała `EXCECPTION_CONTINUE_SEARCH` jest używana w wyrażeniu filtru wyjątków.

 Ten kod jest odpowiednikiem chronionego bloku bez bloku obsługi wyjątków, ponieważ blok obsługi nie jest wykonywany.

## <a name="example"></a>Przykład
 Poniższy kod generuje to ostrzeżenie:

```
#include <excpt.h>
#include <stdio.h>

void f (char *pch)
{
  __try
  {
    // assignment might fail
    *pch = 0;
  }
  __except (EXCEPTION_CONTINUE_SEARCH) // warning 6318
  {
    puts("Exception Occurred");
  }
}
```

 Aby poprawić to ostrzeżenie, należy użyć następującego kodu:

```
#include <excpt.h>
#include <stdio.h>
#include <windows.h>

void f (char *pch)
{
  __try
  {
    // assignment might fail
    *pch = 0;
  }
  __except( (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION) ?
                EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )
  {
    puts("Access violation");
  }
}
```

## <a name="see-also"></a>Zobacz też
 [try-except, instrukcja](/cpp/cpp/try-except-statement)